src/cljs/witan/ui/components/bundle-add.cljs

```diff
@@ -0,0 +1,106 @@
+                                               :kixi.datastore.metadatastore/provenance)
+                                  :title (get-string :string/file-uploaded-at)
+                                  :weight 0.20}])}
+           {:exclusions @files}]
+          (when-not (empty? @files)
+            [shared/table
+             {:headers [{:content-fn
+                         #(vector
+                           :div.flex-start
+
+                           (shared/button {:icon icons/search
+                                           :id (str (:kixi.datastore.metadatastore/id %) "-open")
+                                           :prevent? true}
+                                          (fn [_]
+                                            (.open
+                                             js/window
+                                             (str "/#" (route/find-path :app/data {:id (:kixi.datastore.metadatastore/id %)})))))
+                           (shared/button {:icon icons/delete
+                                           :id (str (:kixi.datastore.metadatastore/id %) "-delete")}
+                                          (fn [_] (swap! files disj %))))
+                         :title ""  :weight "100px"}
+                        {:content-fn #(shared/inline-file-title % :small :small)
+                         :title (get-string :string/file-name)
+                         :weight 0.6}
+                        {:content-fn #(js/filesize (:kixi.datastore.metadatastore/size-bytes %))
+                         :title (get-string :string/file-size)
+                         :weight 0.15}
+                        {:content-fn #(or (get-in
+                                           %
+                                           [:kixi.datastore.metadatastore.license/license
+                                            :kixi.datastore.metadatastore.license/type]) (get-string :string/na))
+                         :title (get-string :string/license)
+                         :weight 0.15}
+                        ]
+              :content @files}])]]))))
+
+
+(defn view
+  []
+  (let [files (r/atom #{})]
+    (fn []
+      [:div#create-datapack-view
+       (shared/header :string/share-files-to-datapack nil #{:center})
+       [:div.flex-center
+        [:div.container.padded-content
+         [editable-field nil
+          [:div
+           (get-string :string/datapack-collect-intro-text)]]
+         [show-files files]
+         (shared/button {:id :datapack-add
+                         :txt :string/datapack-sharing-bundle-add
+                         :class "btn-success"
+                         :prevent? true
+                         :disabled? (empty? @files)}
+                        #(controller/raise! :data/add-collect-files-to-datapack {:added-files @files
+                                                                                 :datapack-id (:id (:route/params (data/get-app-state :app/route)))}))]]])))
```

Comment:
The whole form needs disabling whilst the activity is in progress, to stop people hammering the button, changing files etc. Also, a message if/when it fails/succeeds.